embed_grader/serapis/models.py

```diff
@@ -224,7 +224,7 @@ def viewing_scope_by_user(self, user):
 
         # students cannot see test input if homework is not released yet
         now = timezone.now()
-        if now < self.release_time:
+        if now < self.release_time or (not user.has_perm('view_assignment', self.course_fk)):
```

Comment:
`or not user.has_perm('view_assignment', self.course_fk)`

I think in python we try not to put brackets if it is possible to make it look more like natural language :)